Added rescale on shift+mouse move

bobpepin opened the pull request:
Quick and dirty implementation, when holding down the shift key in edit mode and moving the mouse left/right it will scale the radius of the selected nodes.

huxingyi commented on src/skeletongraphicswidget.h:
```diff
@@ -724,6 +724,8 @@ private slots:
     std::set<QUuid> m_deferredRemoveNodeIds;
     std::set<QUuid> m_deferredRemoveEdgeIds;
+    QPoint m_lastScalePos;
+    bool m_scaleStarted;
```
m_scaleStarted should be initialized as false

huxingyi commented on src/skeletongraphicswidget.cpp:
```diff
@@ -910,6 +910,36 @@ bool SkeletonGraphicsWidget::mouseMove(QMouseEvent *event)
     
+    if (SkeletonDocumentEditMode::Select == m_document->editMode) {
+        if (QGuiApplication::queryKeyboardModifiers().testFlag(Qt::ShiftModifier)) {
+            if(m_scaleStarted) {
```
Add a space after if

huxingyi commented on src/skeletongraphicswidget.cpp:
```diff
@@ -910,6 +910,36 @@ bool SkeletonGraphicsWidget::mouseMove(QMouseEvent *event)
+                    } else {
+                        zoomSelected(delta);
+                    }
+                    emit groupOperationAdded();
```
emit groupOperationAdded only when the scale is done to avoid every fine move operation insert into the undo queue.

bobpepin commented:
Addressed your comments, but I just noticed that shift is already taken for rotations. Seems like we are out of modifier keys... This is good enough for me right now to make the software useable, but you probably don't want to merge it into master at this point.

huxingyi commented:
Oh, we need to figure out a way to avoid conflict.

hikari-no-yume commented:
Am I understanding correctly that this would provide an alternative to the scroll wheel for resizing a node? Right now I am using a mouse with a scroll wheel that doesn't work properly, so it makes using Dust3D harder. Also, even with a scroll wheel, moving the mouse can be better for some things.

bobpepin commented:
Correct. The main reason this has not been committed is that shift+mouse is already used for something else and we’re out of modifier keys. 

> On 24 Oct 2020, at 14.25, Andrea <notifications@github.com> wrote:
> 
> ﻿
> Am I understanding correctly that this would provide an alternative to the scroll wheel for resizing a node? Right now I am using a mouse with a scroll wheel that doesn't work properly, so it makes using Dust3D harder. Also, even with a scroll wheel, moving the mouse can be better for some things.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

huxingyi commented:
It's too old, let's close this one, feel free to raise another one if still interested. thanks.
